beacon-chain/sync/initial-sync/blocks_fetcher_peers.go

```diff
@@ -118,3 +120,39 @@ func (f *blocksFetcher) filterPeers(peers []peer.ID, peersPercentage float64) ([
 
 	return peers, nil
 }
+
+// filterScoredPeers returns transformed list of peers,
+// weight sorted by scores and capacity remaining.
+func (f *blocksFetcher) filterScoredPeers(ctx context.Context, peers []peer.ID, ratio float64) ([]peer.ID, error) {
+	ctx, span := trace.StartSpan(ctx, "initialsync.filterScoredPeers")
+	defer span.End()
+
+	if len(peers) == 0 {
+		return peers, nil
+	}
+
+	// Sort peers using both block provider score and, custom, capacity based score (see
+	// peerFilterCapacityWeight if you want to give different weights to provider's and capacity
+	// scores).
+	// Scores produced are used as weights, so peers are ordered probabilistically i.e. peer with
+	// a higher score has higher chance to end up higher in the list.
+	scorer := f.p2p.Peers().Scorers().BlockProviderScorer()
+	peers = scorer.WeightSorted(f.rand, peers, func(peerID peer.ID, blockProviderScore float64) float64 {
+		remaining, capacity := float64(f.rateLimiter.Remaining(peerID.String())), float64(f.rateLimiter.Capacity())
+		// When no capacity for a good peer left, allow less performant peer to take a chance.
+		if remaining < float64(f.blocksPerSecond) {
+			return 0.0
+		}
+		capScore := remaining / capacity
+		overallScore := blockProviderScore*(1.0-f.capacityWeight) + capScore*f.capacityWeight
+		return math.Round(overallScore*scorers.ScoreRoundingFactor) / scorers.ScoreRoundingFactor
+	})
+
+	// Weak/slow peers will be pushed down the list and trimmed since only percentage of peers is selected.
+	limit := uint64(math.Round(float64(len(peers)) * ratio))
```

Comment:
```go
	limit := uint64(math.Round(float64(len(peers)) * ratio))
	limit = mathutil.Max(limit, uint64(flags.Get().MinimumSyncPeers))
	limit = mathutil.Min(limit, uint64(len(peers)))
```
This whole chunk of code maybe needs more explanation , my main question is what exactly this does
```go
	limit := uint64(math.Round(float64(len(peers)) * ratio))
```
why does it need to be done by a ratio ?

Reply:
That param was called `peersPercentage `, renaming it to `ratio` without adding more comments wasn't very prudent indeed. Will change name back to `peersPercentage`, and add comment that this param allows to limit how many of sorted peers caller wants to be returned.